### PR TITLE
Add retry during extraction

### PR DESCRIPTION
**Pull request by deepsm007**
/cc @openshift/test-platform

Related to the `unexpected EOF`  errors  
https://redhat-internal.slack.com/archives/CBN38N3MW/p1730198582759599

### CODE REVIEW

File: pkg/steps/release/create_release.go [hunk 1]
```diff
@@ -204,7 +204,19 @@ done
 if [[ "$exit_code" != "0" ]]; then
 	exit $exit_code
 fi
-oc adm release extract --from=%q --to=${ARTIFACT_DIR}/release-payload-%s
+for ((i=1; i<=5; i++)); do
+	if oc adm release extract --from=%q --to=${ARTIFACT_DIR}/release-payload-%s; then
```
Comment (stbenjam) [inline pkg/steps/release/create_release.go:208]: Do you know if it's a problem on retry if `${ARTIFACT_DIR}/release-payload-%s` still has contents in it?

**Review by stbenjam — COMMENTED** [review]
Looks useful to get some debug data. Not the only place we run `oc adm release extract`, though.

File: pkg/steps/release/create_release.go (hunk 1 above)
Comment (deepsm007) [reply pkg/steps/release/create_release.go:208]: I tried running it locally with partial contents and there was no error

**Comment by deepsm007** [comment]
/test e2e

breaking-changes test is failing because of [this](https://github.com/openshift/ci-tools/pull/4261)

**Pull request merged**